lib/panoramix/query.ex

```diff
@@ -293,6 +293,13 @@ defmodule Panoramix.Query do
         {filter, nil} ->
           # Likewise
           filter
+        # If either or both filter is an AND already, merge them together
```

Comment:
Is it possible for a children of an AND filter to contain negation (NOT) inside of `fields`?

Reply:
Yes, something like:

    dimensions.foo not in ["x", "y"] and dimensions.bar not in ["a", "b"]

Those are represented as `not` filters each containing (and negating) an `in` filter.

Reply:
I'm thinking of an edge case where a negation might get applied incorrectly after merging. E.g. if intended filters hierarchy is like this:
```
{AND, condition1, {AND, condition2, {NOT, condition3}}}
```
we might end up with
```
{AND, condition1, condition2, {NOT, condition3}}}
```
which will have a different effect since `condition3` will be subtracted from union of 1 and 2, rather than only 2?

Reply:
I think the behaviour would be the same in both cases: the first two conditions must be true and the third condition must be false for the row to be included in the result.